### ISSUE TITLE
Pixel graphics move vertically undesirably

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#1400]
The `pixel` PoC ought have WarMECH march across a field of pink 'a's, and that's exactly what it did until I added the `ncplane_destroy()` on each iteration. Now WarMECH moves down with each iteration, unexpectedly. In `kitty`, the old image is destroyed iff i put a `a=d` in front of the EGC:

```
--- src/lib/kitty.c
+++ src/lib/kitty.c
@@ -78,7 +78,7 @@ write_kitty_data(FILE* fp, int linesize, int leny, int lenx, const uint32_t* dat
 //fprintf(stderr, "total: %d chunks = %d, s=%d,v=%d\n", total, chunks, lenx, leny);
   while(chunks--){
     if(totalout == 0){
-      fprintf(fp, "\e_Gf=32,s=%d,v=%d,a=T%s;", lenx, leny, chunks > 1 ? ",m=1" : "");
+      fprintf(fp, "\e_Ga=d\e\\\e_Gf=32,s=%d,v=%d,a=T%s;", lenx, leny, chunks > 1 ? ",m=1" : "");
     }else{
       fprintf(fp, "\e_Gm=%d;", chunks ? 1 : 0);
     }
```

without this, we still move down, but don't erase the old image (we are thus like sixel).

WarMECH should not be moving down, and should not be leaving a trace behind, in either solution.

**Comment by dankamongmen** [comment]
just to be clear, without the `ncplane_destroy()`, the WarMECH stays on the same line. in `kitty`, without the `ncplane_destroy()` and *with* the `a=d`, we get essentially the correct behavior.

**Comment by dankamongmen** [comment]
for sixel, this comes down to damaging all cells the sprixel moves from/into.

for kitty, this is significantly more complicated. it gives us powerful capabilities regarding displaying a sprite at various locations and removing them, but our model doesn't really match that -- we blit the sprite, and then work as cells.

though, not really...we don't work with them as cells, so much as uncomfortably jam them into the `nccell` framebuffer model. what if we kept some per-cell renderstate across rasterization calls? where we mark every cell that a sprite touches? and we manually damage those at the start of a rasterization pass. then when we print a sprite -- which we always will, since we invalidated it -- we manually mark *undamaged* all cells that it touches (except those above us, which we must somehow identify). these are then exactly the cells we mark damaged at the end. that ought work for sixel.

the same ought work for kitty if we through an `a=d` at the front of every rasterization.

this is supremely inefficient, though -- it forces us to rewrite each sprite every time we rasterize, even if we didn't touch the sprite at all. we *have* to avoid that, i think. though maybe not. think of `raptotheosis`. that's showing a different sprite each time we move, basically -- they're never being carried across rasterizations. so that's optimal. but if you *are* carrying them across...ugh.

currently, what happens, exactly? `pixels` has different behavior for the first two iterations and the following ones. it only advances on the third and subsequent iterations.

* you hit the upper-left corner of the sprite
* you emit the sprite EGC
* you advance to the end of the sprite EGC on the first row, skipping over the breadth of the sprite (this is why we get the black "holes" in our background on the first few columns, and no more, ahh)
* you get to the second row of the sprite.
  * you're damaged wrt text: you're printed
  * you're undamaged wrt text: you're skipped

so in the first iteration, we print WarMech, then we print an 'a' everywhere but the top row, across the breadth of the sprite. on the second iteration, we have damaged the upper left cell, and so it is redrawn. on subsequent iterations, we print no characters, just the sprite...no, that's not right -- we print the invalidated character. interestingly, we print 2 in xterm, and 3 in kitty. they follow the sprite downward. what's going on?
